util/chunk/disk_test.go

```diff
@@ -83,6 +83,51 @@ func (s *testChunkSuite) TestListInDisk(c *check.C) {
 	}
 }
 
+func (s *testChunkSuite) TestListInDiskAppendRow(c *check.C) {
+	numChk, numRow := 2, 2
+	chks, fields := initChunks(numChk, numRow)
+	l := NewListInDisk(fields)
+	for _, chk := range chks {
+		it := NewIterator4Chunk(chk)
+		for row := it.Begin(); row != it.End(); row = it.Next() {
+			ptr, err := l.AppendRow(row)
+			c.Assert(err, check.IsNil)
+			row2, err := l.GetRow(ptr)
+			c.Assert(err, check.IsNil)
+			c.Assert(row.GetDatumRow(fields), check.DeepEquals, row2.GetDatumRow(fields))
+		}
+	}
+	c.Assert(l.NumChunks(), check.Equals, 1)
+	err := l.Add(chks[0])
+	c.Assert(err, check.IsNil)
+	for _, chk := range chks[1:] {
+		it := NewIterator4Chunk(chk)
+		for row := it.Begin(); row != it.End(); row = it.Next() {
+			ptr, err := l.AppendRow(row)
+			c.Assert(err, check.IsNil)
+			row2, err := l.GetRow(ptr)
+			c.Assert(err, check.IsNil)
+			c.Assert(row.GetDatumRow(fields), check.DeepEquals, row2.GetDatumRow(fields))
+		}
+	}
+	c.Assert(l.NumChunks(), check.Equals, 2)
+	c.Assert(l.Len(), check.Equals, numChk*numRow*2)
+
+	records := NewList(fields, numRow, numRow)
+	for _, chk := range chks {
+		records.Add(chk)
+	}
+	for _, chk := range chks {
+		records.Add(chk)
+	}
+	rc := NewRowContainerFromListInDisk(l)
+	it := NewIterator4RowContainer(&rc)
+	it2 := NewIterator4List(records)
+	for row, row2 := it.Begin(), it2.Begin(); row != it.End() && row2 != it2.End(); row, row2 = it.Next(), it2.Next() {
+		c.Assert(row.GetDatumRow(fields), check.DeepEquals, row2.GetDatumRow(fields))
+	}
```

Comment:
What is this test for?

Reply:
All right, it is redundant, since the iterator has already be tested in iterator_test.go, I'll consider remove it.